smac/intensification/successive_halving.py

```diff
@@ -26,6 +26,25 @@ class SuccessiveHalving(Intensifier):
     Implementation from "BOHB: Robust and Efficient Hyperparameter Optimization at Scale" (Falkner et al. 2018)
     (refer supplementary)
 
+    Successive Halving intensifier can operate on two kinds of budgets:
+    1. **'Instances' as budget**:
+        When multiple instances are provided or when run objective is "runtime", this is the criterion used as budgets
+        for successive halving iterations i.e., for each budget, the challengers are tested across an
+        increasing number of instances and their combined performance across all instances is used for evaluation.
```

Comment:
Could you please explicitly note that in every iteration one instance is added and the users have to make sure themselves that this actually doubles the time it takes in practice?

@mlindauer is this the expected behavior? Should we add a mode where the number of instances are explicitly doubled - following more the BOHB implementation?

Reply:
The number of instances actually doubles (increases `eta` folds to be precise) like how it is described in the algorithm. We don't do anything different with increasing budgets for runtime or instances. I'll make it clearer in the docstring.

Reply:
Could you please add to the docs that the number of instances is multiplied by eta?